### ISSUE TITLE
quest

### ISSUE THREAD

**Issue by zhougengx** [JanusGraph/janusgraph#1759]
when start  janusgraph  cassandra.log has these error; but janusgraph start sucess ,when contect has  check the client configuration and connectivity to the server if this message persists
```
CassandraDaemon.java (line 195) Exception in thread Thread[OptionalTasks:1,5,main]
java.lang.NoSuchMethodError: com.codahale.metrics.Snapshot: method <init>()V not found
	at org.apache.cassandra.metrics.DecayingEstimatedHistogramReservoir$EstimatedHistogramReservoirSnapshot.<init>(DecayingEstimatedHistogramReservoir.java:353)
	at org.apache.cassandra.metrics.DecayingEstimatedHistogramReservoir.getSnapshot(DecayingEstimatedHistogramReservoir.java:224)
	at com.codahale.metrics.Histogram.getSnapshot(Histogram.java:54)
	at com.codahale.metrics.Timer.getSnapshot(Timer.java:142)
	at org.apache.cassandra.db.ColumnFamilyStore$3.run(ColumnFamilyStore.java:446)
	at org.apache.cassandra.concurrent.DebuggableScheduledThreadPoolExecutor$UncomplainingRunnable.run(DebuggableScheduledThreadPoolExecutor.java:118)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
ERROR [OptionalTasks:1] 2019-08-23 02:03:10,544 CassandraDaemon.java (line 195) Exception in thread Thread[OptionalTasks:1,5,main]
java.lang.NoSuchMethodError: com.codahale.metrics.Snapshot: method <init>()V not found
	at org.apache.cassandra.metrics.DecayingEstimatedHistogramReservoir$EstimatedHistogramReservoirSnapshot.<init>(DecayingEstimatedHistogramReservoir.java:353)
	at org.apache.cassandra.metrics.DecayingEstimatedHistogramReservoir.getSnapshot(DecayingEstimatedHistogramReservoir.java:224)
	at com.codahale.metrics.Histogram.getSnapshot(Histogram.java:54)
	at com.codahale.metrics.Timer.getSnapshot(Timer.java:142)
	at org.apache.cassandra.db.ColumnFamilyStore$3.run(ColumnFamilyStore.java:446)
	at org.apache.cassandra.concurrent.DebuggableScheduledThreadPoolExecutor$UncomplainingRunnable.run(DebuggableScheduledThreadPoolExecutor.java:118)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by farodin91** [comment]
It looks like a wrong version of codahale. Could you provide more information about your setup? JanusGraph version...

**Comment by zhougengx** [comment]
the least version 0.4.0

**Comment by zhougengx** [comment]
> It looks like a wrong version of codahale. Could you provide more information about your setup? JanusGraph version...

The latest version of  0.4.0

**Comment by farodin91** [comment]
Cassandra requires io.dropwizard.metrics version 3.1 and only ship version 3.3. So, you can start Cassandra as an external version, see here https://docs.janusgraph.org/latest/cassandra.html#_remote_server_mode. 

**Note:** We are on dropping Cassandra Thrift with the upcoming version.  If you are using `janusgraph.sh`, will be start externally like Elasticsearch.

**Comment by captify-ofedorchuk** [comment]
> Cassandra requires io.dropwizard.metrics version 3.1 and only ship version 3.3. So, you can start Cassandra as an external version, see here https://docs.janusgraph.org/latest/cassandra.html#_remote_server_mode.
> 
> **Note:** We are on dropping Cassandra Thrift with the upcoming version. If you are using `janusgraph.sh`, will be start externally like Elasticsearch.

Hi! I have this problem when i try to launch https://github.com/JanusGraph/janusgraph/releases/tag/v0.5.0 . And the link you provided is broken. Can you provide a bit more info on how to cope with this problem?

**Comment by farodin91** [comment]
Due to delay, we didn' t finished shipping cassandra as external depencency. 

Update doc: https://docs.janusgraph.org/storage-backend/cassandra/#remote-server-mode

**Comment by captify-ofedorchuk** [comment]
Have the same problem with 0.5.1 version. Could you, please, provide a more extensive description of how to cope with this problem? I don't realy understand how the link you provided is related to the bug :)

**Comment by captify-ofedorchuk** [comment]
> Due to delay, we didn' t finished shipping cassandra as external depencency.
> 
> Update doc: https://docs.janusgraph.org/storage-backend/cassandra/#remote-server-mode

Have the same problem with 0.5.1 version. Could you, please, provide a more extensive description of how to cope with this problem? I don't realy understand how the link you provided is related to the bug :)

**Comment by vishalgoel1988** [comment]
Hi,

Does this get resolved anyhow? We are seeing the same issue in Janus 0.5.2 with Embedded Cassandra. Anyexternal dependency/Jar that we can add to setup to resolve this?

**Comment by bkusumakar** [comment]
Following the steps mentioned [here](https://github.com/JanusGraph/janusgraph/issues/2058#issuecomment-645993664) resolved this in `janusgraph-full-0.5.2`

**Comment by kusumakarb** [comment]
Following the steps mentioned [here](https://github.com/JanusGraph/janusgraph/issues/2058#issuecomment-645993664) resolved this in `janusgraph-full-0.5.2`
